File: JAVA/2024-12-19/12973.java
//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
import java.util.*;

public class Main {
    public static void main(String[] args) {
    }
}
    class Solution {
        public int solution(String s)
        {
            int answer = -1;
            Stack<Character> stack_char = new Stack<>();

            for (int i = 0; i < s.length(); i++)
            {
                char sc = s.charAt(i);
                if (!stack_char.isEmpty() && stack_char.peek() == sc)
                {
                    stack_char.pop();
                    continue;
                }
                stack_char.push(sc);
            }

            if (stack_char.isEmpty()) answer = 1;
            else answer = 0;

            return answer;
    }
}

File: C++/2024-12-19/12973.cpp
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
	stack<char> stk;

	for (int i = 0; i < s.size(); i++)
	{
		if (!stk.empty() && stk.top() == s[i])
		{
			stk.pop();
			continue;
		}

		stk.push(s[i]);
	}

    return stk.empty();
}
